pkg/archive/archive.go

```diff
@@ -13,7 +13,7 @@ import (
 	"github.com/pkg/errors"
 )
 
-var AllStabilizers = slices.Concat(AllZipStabilizers, AllTarStabilizers, AllGzipStabilizers)
+var AllStabilizers = slices.Concat(AllZipStabilizers, AllTarStabilizers, AllGzipStabilizers, AllJarStabilizers)
```

Comment:
I think i'm feeling some hesitance adding the JAR stabilizers here because these would be the first stabilizers to be definitively ecosystem-specific and the existing uses aren't yet able to differentiate.

The simple solution would be to add ecosystem-mapping somewhere (probably at a higher level than the archive package) that could be used from our provenance generation logic. For the stabilizer CLI, we could infer the ecosystem based off of the file extension and, when ambiguous, require the user explicitly pass in an ecosystem flag.

Apologies for creating more work but I think we're asking for unexpected consequences without that protection.

Do you feel up to making those changes? otherwise, I can take a crack at them and unblock you here.

Reply:
okay I went ahead and fixed this in #418. I think it should be safer to submit this once that's in.

Reply:
That's a good point. I agree with you. I will take a look at that PR. Thanks!

Reply:
It's now merged so this should be safe to rebase onto. You can also make use of the new AllJarStabilizers constant in that new ecosystem-specific logic.

Reply:
Sounds great. I will take a look soon.